subcommand/get-consul-client-ca/command_test.go

```diff
@@ -0,0 +1,388 @@
+		c.Connect = map[string]interface{}{
+			"enabled": true,
+		}
+		c.CAFile = caFile
+		c.CertFile = certFile
+		c.KeyFile = keyFile
+		c.Ports = &testutil.TestPortConfig{
+			DNS:     randomPorts[0],
+			HTTP:    randomPorts[1],
+			HTTPS:   8501,
+			SerfLan: randomPorts[2],
+			SerfWan: randomPorts[3],
+			Server:  randomPorts[4],
+		}
+	})
+	require.NoError(t, err)
+	defer a.Stop()
+
+	// run the command
+	exitCode := cmd.Run([]string{
+		"-server-addr", "provider=fake",
+		"-tls-server-name", "localhost",
+		"-output-file", outputFile.Name(),
+		"-ca-file", caFile,
+	})
+	require.Equal(t, 0, exitCode, ui.ErrorWriter.String())
+
+	// check that the provider has been called
+	require.Equal(t, 1, provider.addrsNumCalls, "provider's Addrs method was not called")
+
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPSAddr,
+		Scheme:  "https",
+		TLSConfig: api.TLSConfig{
+			CAFile: caFile,
+		},
+	})
+	require.NoError(t, err)
+
+	// get the actual root ca cert from consul
+	roots, _, err := client.Agent().ConnectCARoots(nil)
+	require.NoError(t, err)
+	require.NotNil(t, roots)
+	require.NotNil(t, roots.Roots)
+	require.Len(t, roots.Roots, 1)
+	require.True(t, roots.Roots[0].Active)
+	expectedCARoot := roots.Roots[0].RootCertPEM
+
+	// read the file contents
+	actualCARoot, err := ioutil.ReadFile(outputFile.Name())
+	require.NoError(t, err)
+	require.Equal(t, expectedCARoot, string(actualCARoot))
+}
+
+// generateCA generates Consul CA
+// and returns cert and key as pem strings.
+func generateCA(t *testing.T) (caPem, keyPem string) {
+	require := require.New(t)
+
+	sn, err := tlsutil.GenerateSerialNumber()
+	require.NoError(err)
+
+	var signer crypto.Signer
+	signer, keyPem, err = tlsutil.GeneratePrivateKey()
+	require.NoError(err)
+
+	constraints := []string{"consul", "localhost"}
+	caPem, err = tlsutil.GenerateCA(signer, sn, 1, constraints)
+	require.NoError(err)
+
+	return
+}
+
+// generateServerCerts generates Consul CA
+// and a server certificate and saves them to temp files.
+// It returns file names in this order:
+// CA certificate, server certificate, and server key.
+// Note that it's the responsibility of the caller to
+// remove the temporary files created by this function.
+func generateServerCerts(t *testing.T) (string, string, string, func()) {
+	require := require.New(t)
+
+	caFile, err := ioutil.TempFile("", "ca")
+	require.NoError(err)
+
+	certFile, err := ioutil.TempFile("", "cert")
+	require.NoError(err)
+
+	certKeyFile, err := ioutil.TempFile("", "key")
+	require.NoError(err)
+
+	// Generate CA
+	sn, err := tlsutil.GenerateSerialNumber()
+	require.NoError(err)
+
+	s, _, err := tlsutil.GeneratePrivateKey()
+	require.NoError(err)
+
+	constraints := []string{"consul", "localhost"}
+	ca, err := tlsutil.GenerateCA(s, sn, 1, constraints)
+	require.NoError(err)
+
+	// Generate Server Cert
+	name := fmt.Sprintf("server.%s.%s", "dc1", "consul")
```

Comment:
This seems like it's not necessary. Maybe just use `server.dc1.consul` as the `name` below?